ui/app/adapters/totp.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { resolve, allSettled } from 'rsvp';
+import { isEmpty } from '@ember/utils';
+import ApplicationAdapter from './application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+
+export default ApplicationAdapter.extend({
```

Comment:
Modernize this adapter to match current Ember patterns, confirm each method is actually necessary (and not just copy pasta)